stdlib/public/Concurrency/ExecutorAssertions.swift

```diff
@@ -86,7 +86,7 @@ func preconditionTaskOnActorExecutor(
   // TODO: offer information which executor we actually got
   precondition(expectationCheck,
       // TODO: figure out a way to get the typed repr out of the unowned executor
-      "Incorrect actor executor assumption; Expected '\(actor.unownedExecutor)' executor. \(message())",
+      "Incorrect actor executor assumption; Expected '\(_executorGetTypeName(actor.unownedExecutor.executor))' executor, but was executing on '\(_executorGetTypeName(_executorGetCurrentSerialExecutorRef()))'. \(message())",
```

Comment:
This is a mouthful, but the actual question about the implementation I have is...

Should we be reusing `precondition` like this, or try to reimplement its semantics?
It has fairly complex "how to fail" logic, so I think reusing it might be for the best, right?